### PR TITLE
Fixes for fixed prices tests: stricter constructor and separate eth

### PR DESCRIPTION
**Pull request by jflatow**

### CODE REVIEW

File: contracts/Uniswap/UniswapAnchoredView.sol [hunk 1]
```diff
@@ -79,11 +79,16 @@ contract UniswapAnchoredView is UniswapConfig {
...
-            newObservations[uniswapMarket].acc = cumulativePrice;
+            if (uniswapMarket != address(0)) {
+                require(config.priceSource == PriceSource.REPORTER, "only reported prices utilize an anchor");
+                uint cumulativePrice = currentCumulativePrice(config);
+                oldObservations[uniswapMarket].timestamp = block.timestamp;
+                newObservations[uniswapMarket].timestamp = block.timestamp;
+                oldObservations[uniswapMarket].acc = cumulativePrice;
+                newObservations[uniswapMarket].acc = cumulativePrice;
+            } else {
+                require(config.symbolHash != ethHash, "ETH must have a uniswap market specified");
```
Comment (maxwolff) [inline contracts/Uniswap/UniswapAnchoredView.sol:90]: what about FIXED_USD markets? which have no uniswapMarket but arent ETH?

Comment (antonina-cherednichenko) [inline contracts/Uniswap/UniswapAnchoredView.sol:90]: why only eth ? what about other pairs?

Comment (jflatow) [reply contracts/Uniswap/UniswapAnchoredView.sol:90]: good point, not sure why I did that

Comment (jflatow) [reply contracts/Uniswap/UniswapAnchoredView.sol:90]: Those don't need a uniswap market right? They would match not eth and would not revert

**Pull request merged**